### DISCUSSION TITLE
Our usage of <slot> element

### DISCUSSION THREAD

**Discussion by WebReflection** [pyscript/pyscript#1295] [Q&A] [answered]
I am going through the code and wondering what is the idea, or background, behind the usage of slot elements in our components. In case it helps anyone, [slots are not meant to be generic containers](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_templates_and_slots), these have a meaning when templates are used (we don't use template elements) and when either the content is propagated via cloning or slots are named.

In our case it looks like we create slots for no particular reason and I wonder if we could simplify some constructor dance by dropping these unless we were planning to use templates but for distributable components that doesn't look like beneficial for our components' consumers.

Can anyone please clarify if cleaning up a bit is a good thing to do or if there are reasons behind current slots? Thanks!

**Comment by antocuni** [comment, marked as answer]
Pinging @fpliger here, as he is the original author of most of the code related to shadowroot/slots in our components.

**Reply by WebReflection** [reply]
Thanks! For the time being, I've removed the unused / unnecessary slot from *py-repl* https://github.com/pyscript/pyscript/pull/1296 just in case there were more thoughts behind ... but even in other components we create a slot to then append things in it ... that *shouldn't* be needed once `this.shadowRoot` is created via `attachShadow` so I am curious to know if it's OK to simplify those trees.

**Reply by fpliger** [reply]
Thank you @antocuni for the ping and @WebReflection for bringing that up!

Honestly, they are there because of a mix of me partially misunderstanding slots some of it at the time and because of me trying to experiment how to implement some things (i.e., py-repl generating the output and auto& generating a new repl). 

It's fair to say we should remove 👍🙂
